integration/client.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2023 IndyKite
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integration
```

Comment:
The integration folder is only for integration tests: there is no real code in it, only some methods to avoid repetitive code in the test case.
Do you have an idea about how to improve the structure?

Reply:
Well, everything which doesn't have `_test.go` suffix is real code, which can be included and used.
So if some developer will include our SDK, they can use all your code from `integration` package. Including `constants.go`. All those user can include and use in their code. Which is nonsence.

Everything from `authorization_test.go` should be in `authorization` package and from `ingest_test.go` should be in `ingest` package.

In Go tests are always just next to source code in files with `_test.go` suffix. In Go you don't create extra folder just for tests.

Reply:
If you want some "global" test helpers, like what is in constants.go etc use the root `test` folder.

Reply:
updated